### PR TITLE
Endpoint and uniform coloring mode selection

### PR DESCRIPTION
**Pull request by carlOSx64**
- Added new command-line options `--coloring` and `--rgb`.
- Added endpoint and uniform coloring. The new code is mostly contained inside `coloring.rs`.

sheet

### CODE REVIEW

File: src/graphics/pipeline/shaders/streamline.wgsl [hunk 1]
```diff
@@ -14,7 +14,7 @@ struct FragmentInput {
 fn vertex(in: VertexInput) -> FragmentInput {
     return FragmentInput(
         transform * vec4<f32>(in.position, 1.),
-        abs(in.direction)
+        in.direction
```
Comment (nilgoyette) [inline src/graphics/pipeline/shaders/streamline.wgsl]: Why did you remove `abs` here and add it on all callers? It seems simpler (and faster) to have it here.

File: src/graphics/resources/coloring.rs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    fn assign_endpoint(vertices: &mut [FiberVertex], ranges: &[Range<usize>]) {
+        for range in ranges {
+            let (first, last) = (&vertices[range.start], &vertices[range.end]);
+            let color = (first.position - last.position).normalize().abs();
+
+            for i in range.start..=range.end {
+                vertices[i].color = color
+            }
```
Comment (nilgoyette) [inline src/graphics/resources/coloring.rs]: I think you can use [`slice::fill`](https://doc.rust-lang.org/std/primitive.slice.html#method.fill). Below too.

File: src/graphics/pipeline/shaders/streamline.wgsl (hunk 1 above)
Comment (carlOSx64) [reply src/graphics/pipeline/shaders/streamline.wgsl]: The input is now a color and not a direction (I forgot to change the field name here). I removed it since it makes more sense conceptually. But in practice, both approaches work well.

Only the local mode actually needs this because in endpoint mode, the color is processed for each streamline (not vertex), and in uniform mode, there is no processing at all.

Do you recommend putting the function back in or simply renaming the field to `color`?

File: src/graphics/resources/coloring.rs (hunk 2 above)
Comment (carlOSx64) [reply src/graphics/resources/coloring.rs]: Unfortunately, it doesn't seem to work. `slice::fill` directly mutates the slice's content, but here, the values are assigned to a field, adding an indirection.

File: src/graphics/pipeline/shaders/streamline.wgsl (hunk 1 above)
Comment (carlOSx64) [reply src/graphics/pipeline/shaders/streamline.wgsl:17]: `signed_color` would be a fitting name if we use the shader approach.

File: src/graphics/resources/coloring.rs (hunk 2 above)
Comment (nilgoyette) [reply src/graphics/resources/coloring.rs]: Oh, right, sorry.

File: src/graphics/pipeline/shaders/streamline.wgsl (hunk 1 above)
Comment (nilgoyette) [reply src/graphics/pipeline/shaders/streamline.wgsl]: I prefer renaming to wathever you think is right.

**Comment by carlOSx64** [comment]
I'm done, the field was renamed. I also refactored `main.rs`, and I moved some stuff into a new module named `inputs.rs`.
